clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1783,32 +1784,45 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
     Header.startKernel(Name, NameType, StableName, KernelObj->getLocation());
   }
 
+  size_t getParamCount(QualType FieldTy){
+    const auto *SamplerOrAccessorTy = FieldTy->getAsCXXRecordDecl();
```

Comment:
Both places we call this have access to a CXXRecordDecl I think, so this function can just take that instead.